### PR TITLE
Bugfix/table rotation

### PR DESCRIPTION
**Pull request by derekclee**
### Overview
* Fixes a major bug in the code when using a month partition interval.
  For short months, like February, the generated partitions skipped over that month (Feb)
  resulting in that table being dropped because it was not part of the table collection.
  This only occurred at the start of a short month.
* Related HB: https://app.honeybadger.io/projects/43244/faults/32893983

### CODE REVIEW

**Comment by derekclee** [comment]
cc @kapost/shibui @bwebster

File: spec/lib/sliding_partition/table_collection_spec.rb [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+                 time_column: :timestamp,
+                 suffix: "%Y%m%d",
+                 partition_interval: 1.month,
+                 retention_interval: 6.months
+  end
+
+  subject { described_class.new(definition: definition, at_time: at_time) }
+
+  shared_examples_for "a table generator" do
+    it "works" do
```
Comment (bwebster) [inline spec/lib/sliding_partition/table_collection_spec.rb]: :neckbeard: Maybe use a more descriptive description here?

**Review by bwebster — APPROVED** [review]
Looks good

File: spec/lib/sliding_partition/table_collection_spec.rb (hunk 1 above)
Comment (derekclee) [reply spec/lib/sliding_partition/table_collection_spec.rb]: sure.

**Pull request merged**